go-controller/pkg/node/gateway_localnet_linux_test.go

```diff
@@ -99,7 +99,7 @@ func newObjectMeta(name, namespace string) metav1.ObjectMeta {
 }
 
 func newService(name, namespace, ip string, ports []v1.ServicePort, serviceType v1.ServiceType,
-	externalIPs []string, serviceStatus v1.ServiceStatus, isETPLocal, isITPLocal bool) *v1.Service {
+	externalIPs []string, serviceStatus v1.ServiceStatus, isETPLocal, isITPLocal, allocateNPs bool) *v1.Service {
```

Comment:
By doing this you have to update the test code all over the place. I think that makes it sub-optimal.

What do you think about adding a new function that calls this one, but sets the value ?

Then you'd just have to focus on your single test (which will be the only one using this new func) - and reviewers will get a lot less changes to look at.

Something like:
```golang
func newServiceWithoutNodePortAllocation(name, namespace, ip string, ports []v1.ServicePort, serviceType v1.ServiceType,
	externalIPs []string, serviceStatus v1.ServiceStatus, isETPLocal, isITPLocal bool) *v1.Service {

    doNotAllocateNodePorts := false
    service := newService(name, namespace, ip, ports, serviceType, externalIPs, serviceStatus, isETPLocal, isITPLocal)
    service.AllocateLoadBalancerNodePorts = &doNotAllocateNodePorts
    return service
}

func newService(name, namespace, ip string, ports []v1.ServicePort, serviceType v1.ServiceType,
	externalIPs []string, serviceStatus v1.ServiceStatus, isETPLocal, isITPLocal bool) *v1.Service {

    allocateNodePorts := true
    return &v1.Service{
		ObjectMeta: newObjectMeta(name, namespace),
		Spec: v1.ServiceSpec{
			ClusterIP:             ip,
			Ports:                 ports,
			Type:                  serviceType,
			ExternalIPs:           externalIPs,
			ExternalTrafficPolicy: externalTrafficPolicy,
			InternalTrafficPolicy: &internalTrafficPolicy,
			ClusterIP:                     ip,
			Ports:                         ports,
			Type:                          serviceType,
			ExternalIPs:                   externalIPs,
			ExternalTrafficPolicy:         externalTrafficPolicy,
			InternalTrafficPolicy:         &internalTrafficPolicy,
			AllocateLoadBalancerNodePorts: &allocateNodePorts,
		},
		Status: serviceStatus,
	}
}
```

Reply:
I love this suggestion! I admit I have been lazy, infact I will do this for ITP and ETP as well, split them as separate functions that invoke the main newService.

Reply:
done.